### PR TITLE
Widget overview docs

### PR DESCRIPTION
**Pull request by tlambert03**
this is a more curated introduction to widgets and their types.  @jni would be nice to get your thoughts on this one.

### CODE REVIEW

**Review by jni — COMMENTED** [review]
Love this @tlambert03. I've left some super minor comments/suggestions, feel free to use/ignore and then merge when ready. 😊

**Review by jni — APPROVED** [review]
TIL that Cmd-Enter on the review box uses "Comment" rather than "Approve". 😂

File: docs/usage/widget_overview.md [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+Because there are often multiple "valid" widget types for a given python object,
+you may sometimes wish to create widgets directly, or use the `widget_type` argument
+in {func}`~magicgui.widgets.create_widget`
+```
+
+## The widget hierarchy
+
+Many widgets present similar types of information in different ways.  `magicgui` tries
+to maintain a consistent API among all types of widgets that are designed to represent
+similar objects.  The general class of widget you are working with will determine the
```
[the remark this answers is not in the document]
Comment (tlambert03) [reply docs/usage/widget_overview.md:64]: oh, I forgot to mention, I code on a type writer... ;)

(the double space muscle memory is very deep unfortunately ...)

**Comment by tlambert03** [comment]
thanks @jni!  much appreciated

**Pull request merged**